src/predictor/gpu_predictor.cu

```diff
@@ -342,7 +342,7 @@ class GPUPredictor : public xgboost::Predictor {
       Predictor::Predictor{generic_param} {}
 
   ~GPUPredictor() override {
-    if (generic_param_->gpu_id >= 0) {
+    if (generic_param_->gpu_id >= 0 && generic_param_->gpu_id < common::AllVisibleGPUs()) {
```

Comment:
This is necessary because when the GPUID is invalid `cudaSetDevice` causes the JVM to crash.

I am not happy about this but do not see a less intrusive way to fix this.